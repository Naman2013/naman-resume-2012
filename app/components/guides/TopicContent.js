import React from 'react';
import CenterColumn from 'components/common/CenterColumn';
import GuideSection from 'components/guides/GuideSection';
import GuideBodyContent from 'components/guides/GuideBodyContent';
import GuideContentList from 'components/guides/GuideContentList';
import style from './TopicContent.style';

const bodyContent = `Mauris non tempor quam, et lacinia sapien. Mauris accumsan eros eget libero lorem posuere vulputate. Etiam elit elit, elementum sed varius at, adipiscing evitae est. Sed nec felis loren posuere posuere, rutrum eu ipsum. Aliquam eget odio sed ligula dolae iaculis consequat at eget orci. Mauris moleistie sit amet metus loi mass imattis varius Donec sit amet ligula eget nisi sodales lorem a molestie bibendum. Etiam nisi anteni consectetur eget placerat a, tempus a neque. Donec ut elit urna. Etiam venenatis leni eleifend urna eget scelerisqueliquam in nunc.

Donec sit amet ligula eget nisi sodales egestas. Aliquam interdum dolor aliquet dolor  iaculis consequat at eget orci. Mauris moleistie sit amet metus loi mass imattis varius Donec sit amet ligula eget nisi sodales lorem a molestie bibendum. Etiam nisi anteni
posuere vulputate. Etiam elit elit, elementum sed varius at.`;

const stubGuideSectionContent = {
  content: () => <GuideBodyContent title="About this guide" content={bodyContent} />,
  column: () => (<GuideContentList list={['contains 10 objects', 'object type guide', 'beginners and up']} />),
  alignContent: 'left',
};

const TopicContent = () => (
  <div className="root">
    <CenterColumn theme={{ marginBottom: '60px', marginTop: '10px' }}>
      <GuideSection {...stubGuideSectionContent} />
    </CenterColumn>
    <style jsx>{style}</style>
  </div>
);

export default TopicContent;
