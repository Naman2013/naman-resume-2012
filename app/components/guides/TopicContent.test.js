import React from 'react';
import { shallow } from 'enzyme';
import TopicContent from './TopicContent';

describe('TopicContent', () => {
  const shallowWrapper = shallow(<TopicContent />);
  it('should render correctly', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
});
