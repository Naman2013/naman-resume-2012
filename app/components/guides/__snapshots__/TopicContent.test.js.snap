// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicContent should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopicContent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <CenterColumn
          breakpoints={
            Array [
              "screen and (min-width: 768px)",
              "screen and (min-width: 1024px)",
              "screen and (min-width: 1366px)",
            ]
          }
          theme={
            Object {
              "marginBottom": "60px",
              "marginTop": "10px",
            }
          }
          widths={
            Array [
              "620px",
              "780px",
              "940px",
            ]
          }
        >
          <GuideSection
            alignContent="left"
            column={[Function]}
            content={[Function]}
          />
        </CenterColumn>,
        <JSXStyle
          css={
            Array [
              ".root.jsx-2170106365{background:url(https://vega.slooh.com/assets/v4/common/White_Paper_Subtle.png);}",
            ]
          }
          styleId="2170106365"
        />,
      ],
      "className": "jsx-2170106365 root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "breakpoints": Array [
            "screen and (min-width: 768px)",
            "screen and (min-width: 1024px)",
            "screen and (min-width: 1366px)",
          ],
          "children": <GuideSection
            alignContent="left"
            column={[Function]}
            content={[Function]}
          />,
          "theme": Object {
            "marginBottom": "60px",
            "marginTop": "10px",
          },
          "widths": Array [
            "620px",
            "780px",
            "940px",
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alignContent": "left",
            "column": [Function],
            "content": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            ".root.jsx-2170106365{background:url(https://vega.slooh.com/assets/v4/common/White_Paper_Subtle.png);}",
          ],
          "styleId": "2170106365",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <CenterColumn
            breakpoints={
              Array [
                "screen and (min-width: 768px)",
                "screen and (min-width: 1024px)",
                "screen and (min-width: 1366px)",
              ]
            }
            theme={
              Object {
                "marginBottom": "60px",
                "marginTop": "10px",
              }
            }
            widths={
              Array [
                "620px",
                "780px",
                "940px",
              ]
            }
          >
            <GuideSection
              alignContent="left"
              column={[Function]}
              content={[Function]}
            />
          </CenterColumn>,
          <JSXStyle
            css={
              Array [
                ".root.jsx-2170106365{background:url(https://vega.slooh.com/assets/v4/common/White_Paper_Subtle.png);}",
              ]
            }
            styleId="2170106365"
          />,
        ],
        "className": "jsx-2170106365 root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "breakpoints": Array [
              "screen and (min-width: 768px)",
              "screen and (min-width: 1024px)",
              "screen and (min-width: 1366px)",
            ],
            "children": <GuideSection
              alignContent="left"
              column={[Function]}
              content={[Function]}
            />,
            "theme": Object {
              "marginBottom": "60px",
              "marginTop": "10px",
            },
            "widths": Array [
              "620px",
              "780px",
              "940px",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "alignContent": "left",
              "column": [Function],
              "content": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              ".root.jsx-2170106365{background:url(https://vega.slooh.com/assets/v4/common/White_Paper_Subtle.png);}",
            ],
            "styleId": "2170106365",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
